File: Homework/JavaScript/#1 - Fundamentals/#5 - Array/4.MaximalIncreasingSequence/MaximalIncreasingSequence.js
// Write a script that finds the maximal increasing sequence in an array.

function findMaximalIncreasingSequence() {
    var arr = [3, 2, 3, 4, 5, 1, 2, 4],
        i,
        len = 1,
        bestlen = 1;

    for (i = 0; i < arr.length; i++) {
        if ((arr[i] + 1) == arr[i + 1]) {
            len++;
        }
        else {
            len = 1;
        }

        if (bestlen < len) {
            bestlen = len;
        }
    }

    console.log("Length: " + bestlen);
}

File: Homework/C#/Part II/#1 - Arrays/5.MaximalIncreasingSequence/MaximalIncreasingSequence.cs
// Write a program that finds the maximal increasing sequence in an array.
using System;

namespace MaximalIncreasingSequence
{
    class MaximalIncreasingSequence
    {
        static void Main()
        {
            int[] arr = { 3, 2, 3, 4, 5, 6, 2, 4 };
            int len = 1;
            int bestlen = 1;

            for (var i = 0; i < arr.Length - 1; i++)
            {
                if ((arr[i] + 1) == arr[i + 1])
                {
                    len++;
                }
                else if ((arr[i] + 1) != arr[i + 1])
                {
                    len = 1;
                }

                if (len > bestlen)
                {
                    bestlen = len;
                }
            }
            Console.WriteLine("Length: {0}", bestlen);
        }
    }
}
